### ISSUE TITLE
Sketcher: B-spline not constrained on circle, nor tangent

### ISSUE THREAD

**Issue by NathanLKing** [FreeCAD/FreeCAD#18115]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Problem description

I create a circle, create a b spline, constrain the spline to a point, then try to add tangent constraint between the circle and the b spline, it moves and applies. 
Yet, zoom in and it shows that it isn't. Try to pad and it throws the "Pad: Wire is not closed"
Which is shown in Sketcher. 

### Full version info

```shell
OS: Windows 10 build 19045
Architecture: x86_64
Version: 1.0.0.39109 (Git) Conda
Build type: Release
Branch: (HEAD detached at 1.0.0)
Hash: 2fcc5317fe3aee96ca73475986a577719fc78e20
Python 3.11.10, Qt 5.15.15, Coin 4.0.3, Vtk 9.3.0, OCC 7.8.1
Locale: English/Australia (en_AU)
Stylesheet/Theme/QtStyle: unset/FreeCAD Classic/Qt default
Installed mods: 
  * Assembly4 0.50.15
  * fasteners 0.5.27
  * QuickMeasure 2022.10.28
```

### Subproject(s) affected?

Sketcher

### Anything else?

![image](https://github.com/user-attachments/assets/65991af3-dbfe-457c-8235-ca959d245a39)
Attached file
[Heart.zip](https://github.com/user-attachments/files/17897673/Heart.zip)

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by tarman3** [comment]
Please attach file

**Comment by NathanLKing** [comment]
![image](https://github.com/user-attachments/assets/65991af3-dbfe-457c-8235-ca959d245a39)

**Comment by NathanLKing** [comment]
Another issue, this system won't allow attaching a freecad file, what the?

**Comment by tarman3** [comment]
Pack FCStd file to zip

**Comment by NathanLKing** [comment]
> Pack FCStd file to zip

Yes, crazy that I have to do that, especially because 7zip files aren't allowed either, 
[Heart.zip](https://github.com/user-attachments/files/17897673/Heart.zip)
but here it is.

**Comment by tarman3** [comment]
The problem not with tangent. Your wire in the sketch is really not closed.

![Screenshot_20241125_074804](https://github.com/user-attachments/assets/177a77f9-7db3-4125-85bc-677a47ab91bb)

PS You chose a very strange way to construct geometry.

**Comment by NathanLKing** [comment]
Yes, that's the point. Freecad won't close it when I choose to add the tangent constraint.
What's strange about it?

**Comment by tarman3** [comment]
Tangent does not mean the curves are connected.

**Comment by tarman3** [comment]
PS
I strongly recommend replacing bspline with a pair of arcs

**Comment by NathanLKing** [comment]
> Tangent does not mean the curves are connected.
I am aware.
When I add the coincident constraint it still shows wire isn't closed.

**Comment by NathanLKing** [comment]
Ah, I found the problem. 
For some reason the Trim tool wasn't taking all of the lines, trimmed some off and left the rest.

**Comment by NathanLKing** [comment]
Yet, of course, there's another issue. It won't allow edits of the spline.

**Comment by tarman3** [comment]
![Screenshot_20241125_075324](https://github.com/user-attachments/assets/f04834a8-ee6f-40aa-8722-4484c6c0ce34)

**Comment by NathanLKing** [comment]
Exactly.... 
Seems yet another issue occuring when I initiate a spline and connect it to a point via a coincident constraint

**Comment by tarman3** [comment]
Replace bspline with arcs and the problems will solve themselves

**Comment by NathanLKing** [comment]
I'm trying that now.
Doesn't actually solve the FreeCAD errors with splines

**Comment by NathanLKing** [comment]
That's a whole lot of steps to mend the errors created by FreeCAD.

**Comment by tarman3** [comment]
There are many errors created by you))

**Comment by NathanLKing** [comment]
I simply used the constraints that are built into FreeCAD

**Comment by tarman3** [comment]
Over time, experience in optimal problem solving will come

**Comment by NathanLKing** [comment]
Stop with the dribble. You're being arrogant and unhelpful.

**Comment by tarman3** [comment]
I hope my video helped you.
Have a good day.

**Comment by luzpaz** [comment]
OP is asking for UI to guide user to a solution. Any ideas how this can happen ?

**Comment by maxwxyz** [comment]
what is the actual issue here?

**Comment by luzpaz** [comment]
@NathanLKing would you mind succinctly describing what is the issue and what you expect FreeCAD to do ?

**Comment by luzpaz** [comment]
there are a few potential issues in here. @AjinkyaDahale could you assist ?

**Comment by AjinkyaDahale** [comment]
> there are a few potential issues in here. @AjinkyaDahale could you assist ?

Seems the OP has issue working with a combination of B-splines, tangent, and trim. The ideal solution would be to just create an end-to-end tangent in the first place (also consider angle constraints if tangent isn't right). However, if there's a specific issue in trim that needs to be addressed (it is possible the refactor introduced this).

The title of this issue doesn't describe that question though. There is no way to "constrain a B-spline on a circle".

Further discussions of using simpler tools may be valid, but should likely be held outside of this issue. The issue can stay even if the OP decides to move to different methods.

**Comment by luzpaz** [comment]
Thanks. So we need a better name for this ticket and we need to split this ticket in regards to trim and simpler tools.

**Comment by luzpaz** [comment]
@NathanLKing is the trim issue this ?  
* https://github.com/FreeCAD/FreeCAD/issues/18497
